Fix: User pacing test

Sakoram opened the pull request.

DawidWesierski4 commented on lib/src/st2110/pipeline/st20_pipeline_tx.c:
```diff
@@ -984,6 +984,26 @@ int st20p_tx_get_sch_idx(st20p_tx_handle handle) {
   return st20_tx_get_sch_idx(ctx->transport);
 }
 
+int st20p_tx_get_pacing_params(st20p_tx_handle handle, double* tr_offset_ns,
```
please add st40 support 
thank you

DawidWesierski4 reviewed (commented):
Imo looks good but there is a concern with accuracy of timestamping of st40 with the recent changes

DawidWesierski4 commented on include/st_pipeline_api.h:
```diff
@@ -319,6 +319,8 @@ struct st_frame {
    * ST20P_RX_FLAG_TIMING_PARSER_META */
   struct st20_rx_tp_meta* tp[MTL_SESSION_PORT_MAX];
+  /** TAI timestamp measured right after first packet of the frame was received */
+  uint64_t receive_timestamp;
```
If we want to include that, let's include that in every session 

Rest is not applicable, but in st40 and st30 frame modes, this should also be there to minimize the discrepancy between sessions

DawidWesierski4 commented on lib/src/st2110/pipeline/st20_pipeline_rx.c:
```diff
@@ -217,6 +217,8 @@ static int rx_st20p_frame_ready(void* priv, void* frame,
   framebuff->src.rtp_timestamp = framebuff->dst.rtp_timestamp = meta->rtp_timestamp;
   framebuff->src.status = framebuff->dst.status = meta->status;
+  framebuff->src.receive_timestamp = framebuff->dst.receive_timestamp =
+      meta->timestamp_first_pkt;
```
same as above please include the same for st30 and st40, 
may not be used in the end but still

DawidWesierski4 commented on tests/integration_tests/noctx/St20pHandler.cpp:
```diff
@@ -62,7 +68,6 @@ void St20pHandler::fillSt20Ops(uint transmissionPort, uint framebufferQueueSize,
   memcpy(sessionsOpsTx.port.dip_addr[MTL_SESSION_PORT_R], ctx->mcast_ip_addr[MTL_PORT_R],
          MTL_IP_ADDR_LEN);
 
-  /* Don't enable Redundant by default */
```
It's still not enabled by default. Why delete this comment?

DawidWesierski4 replied:
why delete this comment ?

DawidWesierski4 commented on tests/integration_tests/noctx/noctx.md:
my bad but please fix

DawidWesierski4 commented on tests/integration_tests/noctx/handlers/st20p_handler.hpp:
```diff
@@ -0,0 +1,52 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
```
please add copyright ?

Sakoram replied:
fixed

DawidWesierski4 commented on tests/integration_tests/noctx/handlers/st20p_handler.cpp:
```diff
@@ -62,7 +68,6 @@ void St20pHandler::fillSt20Ops(uint transmissionPort, uint framebufferQueueSize,
          MTL_IP_ADDR_LEN);
 
-  /* Don't enable Redundant by default */
   sessionsOpsTx.port.num_port = 1;
```
why delete this comment ?

Sakoram replied:
I added this back

DawidWesierski4 commented on tests/integration_tests/noctx/testcases/st20p_user_pacing_tests.cpp:
```diff
@@ -0,0 +1,119 @@
+      << "st20p_user_pacing did not receive any frames";
+  ASSERT_EQ(frameTestStrategyTx->idx_tx, frameTestStrategyRx->idx_rx)
+      << "TX/RX frame count mismatch";
+}
```
i gennerally think this is overcomplicated, 

why not create just the St20Handler as one element,
add the funcitons that will do the cleanup in the default rx, 
i would generally just modify the st20 default function to do somthing along the line of 

if (stop_flag){
do once
  initalizaiton period (to make sure tx is intialized)

test_function()
}

until (deinitalization_period) {
test_function()
}


I would not use the shared state at all, as the frameTestStrategy would suffice, then it would all be in one element 

If done this way this whole class could be removed. As the frameStrategy or it's child that would work on pacing for you would serve that function without the shared object

Sakoram replied:
addressed. just one handler is used now.

DawidWesierski4 reviewed (commented):
This is huge the comments here are the most important to me, 

imo there are very good qualites of this change 
1 - > classes moved into files that will definitely allow us to do multiple session tests big thing i wanted to do but kinda blocked myself via putting the testsStrategies into test files itself

2 -> more automatic handlers, this improves cleanup for sure 


 i just don't get why add the complication of this shared state in sessions that are meant to work together 

the perfectly good resolution would be just to use it as intended imo

Sakoram commented:
> This is huge the comments here are the most important to me,
> 
> imo there are very good qualites of this change 1 - > classes moved into files that will definitely allow us to do multiple session tests big thing i wanted to do but kinda blocked myself via putting the testsStrategies into test files itself
> 
> 2 -> more automatic handlers, this improves cleanup for sure
> 
> i just don't get why add the complication of this shared state in sessions that are meant to work together
> 
> the perfectly good resolution would be just to use it as intended imo

I removed shared state between strategies and just used 2 sessions in one handler. I just added delay between closing RX, and it works. Thanks.

Sakoram commented on tests/integration_tests/noctx/testcases/st20p_user_pacing_tests.cpp:
```diff
@@ -0,0 +1,119 @@
+      [jitterMultipliers](St20pHandler* handler) {
+        return new St20pUserTimestamp(handler, jitterMultipliers);
+      },
+      nullptr, jitterPacingState);
```
addressed. just one handler is used now.
